### DISCUSSION TITLE
error when using bridge: requestAnimationFrame is not defined

### DISCUSSION THREAD

**Discussion by ajaxe** [quasarframework/quasar#13018] [CLI - BEX mode] [answered]
Hi

I am seeing the following console error when using bex bridge (type GlobalQuasarBex), to send message from the popup to background.js.
```
background.js:550 Uncaught (in promise) ReferenceError: requestAnimationFrame is not defined
    at background.js:550:9
    at new Promise (<anonymous>)
    at Bridge._nextSend (background.js:494:14)
    at Bridge._send (background.js:487:19)
    at Bridge.send (background.js:473:19)
    at background.js:586:16
(anonymous)	@	background.js:550
_nextSend	@	background.js:494
_send	@	background.js:487
send	@	background.js:473
(anonymous)	@	background.js:586
```
The error seems to occur when the background script is invoking `bridge.send(event.eventResponseKey, ...);` to resolve the promise. May be the bridge is using method on "window" object which is not longer available in background.js as it is a service worker.

Use repo https://github.com/ajaxe/quasar-bex-troubleshoot to reproduce the issue. First button shows the error in service work console the second one does not.

**Comment by rstoenescu** [comment, marked as answer]
Excellent point. Thanks for reporting this.
Pushed a fix and it will be available in q/app-vite beta.13
